### PR TITLE
(SIMP-2932) add svckill::mode to simp config for simp scenario

### PR DESCRIPTION
**Pull request by lnemsick-simp**
- Provide mechanism to build each scenario's, decision tree YAML from
  shared parts, to allow code reuse while supporting scenario divergence.
- Add query for svckill::mode to simp scenario item tree (scenario divergence)
- Don't write out simp_options::ldap and simp_options::sssd when they exist in hieradata. 
  (This annoying nit couldn't be easily done prior to the new mechanism.)

SIMP-2932 #close

### CODE REVIEW

File: lib/simp/cli/config/items_yaml_generator.rb [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      part_yaml = IO.read(part_file)
+      part_yaml = make_substitutions(part_yaml, substitutions) unless substitutions.empty?
+      yaml += part_yaml + "\n"
+    end
+    yaml
+  end
+
+  def load_scenario_items_yaml
+    scenario_items_file = File.join(@scenarios_dir, "#{@scenario}_items.yaml")
+puts scenario_items_file
```
Comment (nick-markowski) [inline lib/simp/cli/config/items_yaml_generator.rb]: looks like debugging cruft

File: lib/simp/cli/config/items/data/svckill_mode.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+              down and disabled, without actually making the changes
+              to the system
+
+NOTICE: svckill is the mechanism that SIMP uses to comply with the
+requirement that no unauthorized services are running on your system.
+Is it HIGHLY recommended that you set this to 'enforcing'. Please be
+aware that, if you do this, svckill will stop ALL services that are
+not referenced in your Puppet configuration.}
+
+      @warning_msgs = {
```
Comment (nick-markowski) [inline lib/simp/cli/config/items/data/svckill_mode.rb]: We might consider having users 'press any key to continue' so this does not fly by on the screen after application

File: lib/simp/cli/config/items/data/svckill_mode.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
+require File.expand_path( '../item', File.dirname(__FILE__) )
+
+module Simp; end
+class Simp::Cli; end
+module Simp::Cli::Config
+  class Item::SvckillMode < Item
+    def initialize
+      super
+      @key         = 'svckill::mode'
+      @description = %Q{
```
Comment (nick-markowski) [inline lib/simp/cli/config/items/data/svckill_mode.rb]: The description has an extra line after the === svckill::mode === declaration, which does not follow the format in the rest of simp config.

**Comment by nick-markowski** [comment]
This looks great. I still need to test on a fresh install against poss and lite.

File: lib/simp/cli/config/items_yaml_generator.rb (hunk 1 above)
Comment (lnemsick-simp) [inline lib/simp/cli/config/items_yaml_generator.rb]: will remove

File: lib/simp/cli/config/items/data/svckill_mode.rb (hunk 3 above)
Comment (lnemsick-simp) [inline lib/simp/cli/config/items/data/svckill_mode.rb]: Will fix.

File: lib/simp/cli/config/items/data/svckill_mode.rb (hunk 2 above)
Comment (lnemsick-simp) [inline lib/simp/cli/config/items/data/svckill_mode.rb]: Good idea for a future enhancement for this and similar warnings.  For now, I suggest we stick with the pause.  The user can always go back and review the log.

Comment (nick-markowski) [inline lib/simp/cli/config/items/data/svckill_mode.rb]: +1

**Pull request merged**